### PR TITLE
NIFI-4193 Converting to use the dockerfile-maven plugin 

### PR DESCRIPTION
**Pull request by apiri**
NIFI-4193 Converting to use the dockerfile-maven plugin to replace deprecated plugin.

### CODE REVIEW

**Comment by jfrazee** [comment]
Reviewing for once

File: nifi-docker/dockermaven/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    OF ANY KIND, either express or implied. See the License for the specific
+    language governing permissions and limitations under the License. -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.nifi</groupId>
+        <artifactId>nifi-docker</artifactId>
+        <version>1.4.0-SNAPSHOT</version>
```
Comment (jfrazee) [inline nifi-docker/dockermaven/pom.xml]: @apiri Shouldn't this be 1.5.0-SNAPSHOT now? The PR is into master and not a maint branch so wanted to check.

Comment (apiri) [reply nifi-docker/dockermaven/pom.xml]: Yep.  This was originally submitted a couple months ago.  Will update.

Comment (apiri) [reply nifi-docker/dockermaven/pom.xml]: updated to use ${project.version}

Comment (jfrazee) [reply nifi-docker/dockermaven/pom.xml]: @apiri Hey, your ${project.version} change was just overlaid on top of master so all the dockerfile-maven plugin changes are gone.

**Comment by apiri** [comment]
@jfrazee Indeed... you didn't happen to notice this before updating your pull did you?  I seem to have completely lost it on my local repo (and I'm thinking I compounded issues by picking an untimely point to do a manual gc)

**Comment by apiri** [comment]
well, git reflog has saved me.  will fix this up

**Comment by apiri** [comment]
okay, should actually have the material changes too!  thanks for alerting me.

**Comment by jfrazee** [comment]
LGTM +1

Successfully ran build w/ -Pcontrib-check and tested image w/ `docker run -it -p 8080:8080 apache/nifi:latest`

**Pull request merged**